src/Bicep.Core/Analyzers/Linter/Rules/UseParentPropertyRule.cs

```diff
@@ -0,0 +1,174 @@
+                        TryGetParentName(childName, parentNameSyntax) is {} replacement &&
+                        TryCreateDiagnostic(diagnosticLevel, parentResource, resource, replacement) is {} diagnostic)
+                    {
+                        yield return diagnostic;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<string> RemoveFirstLeadingSlash(IEnumerable<string> segments)
+            => segments.Select((x, i) => i == 0 ? x.Substring(1) : x);
+
+        private SyntaxBase? TryGetParentName(StringSyntax parentName, StringSyntax childName)
+        {
+            /* Handle cases such as:
+             * parent: 'abc', child: 'abc/def'
+             * parent: '${abc}', child: '${abc}/${def}'
+             * parent: '${abc}', child: '${abc}/def'
+             * parent: '${abc}-foo', child: '${abc}-foo/bar'
+            */
+
+            if (childName.SegmentValues.Length < parentName.SegmentValues.Length ||
+                childName.Expressions.Length < parentName.Expressions.Length)
+            {
+                // child name must have more or equal string segments and expressions
+                return null;
+            }
+
+            for (var i = 0; i < parentName.SegmentValues.Length - 1; i++)
+            {
+                if (childName.SegmentValues[i] != parentName.SegmentValues[i])
+                {
+                    // apart from the last child segment, all segments must match those of the parent
+                    return null;
+                }
+            }
+
+            var finalParentSegment = parentName.SegmentValues[parentName.SegmentValues.Length - 1];
+            var finalChildSegment = childName.SegmentValues[parentName.SegmentValues.Length - 1];
+
+            if (finalChildSegment.Length <= finalParentSegment.Length ||
+                finalChildSegment[finalParentSegment.Length] != '/' ||
+                finalChildSegment.IndexOf(finalParentSegment) != 0)
+            {
+                // the last common segment must either match exactly, or be equal to <parent>/<something>
+                return null;
+            }
+
+            if (!Enumerable.SequenceEqual(
+                childName.Expressions.Take(parentName.Expressions.Length),
+                parentName.Expressions,
+                SyntaxIgnoringTriviaComparer.Instance))
+            {
+                // the expressions must all match.
+                //this check is expensive - only run if everything else looks ok.
+                return null;
+            }
+
+            var newString = SyntaxFactory.CreateString(
+                new [] { finalChildSegment.Substring(finalParentSegment.Length + 1) }.Concat(childName.SegmentValues.Skip(parentName.SegmentValues.Length)),
+                childName.Expressions.Skip(parentName.Expressions.Length));
+
+            return SimplifyInterpolationRule.TrySimplify(newString) ?? newString;
```

Comment:
`SimplifyInterpolationRule::TrySimplify` may not return a string:

```bicep
var a = 100

resource ... = {
  name: 'parent'
}

resource ... = {
  name: 'parent/${a}'
}
```

Reply:
Great catch! I've added verification that the value being assigned is assignable to type `string`.